Sources/AuthFoundation/OAuth2/OAuth2ClientConfiguration.swift

```diff
@@ -0,0 +1,113 @@
+//
+// Copyright (c) 2022-Present, Okta, Inc. and/or its affiliates. All rights reserved.
+// The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+//
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and limitations under the License.
+//
+
+import Foundation
+
+extension OAuth2Client {
+    public enum PropertyListConfigurationError: Error {
+        case defaultPropertyListNotFound
+        case invalidPropertyList(url: URL)
+        case cannotParsePropertyList(_ error: Error?)
+        case missingConfigurationValues
+    }
+    
+    /// Utility struct used internally to process `Okta.plist` and other similar client configuration files.
+    ///
+    /// > Important: This struct is intended for internal use, and may be subject to change.
+    public struct PropertyListConfiguration {
+        public let issuer: URL
+        public let clientId: String
+        public let scopes: String
+        public let redirectUri: URL?
+        public let logoutRedirectUri: URL?
+        public let additionalParameters: [String: String]?
+
+        public init() throws {
+            guard let file = Bundle.main.url(forResource: "Okta", withExtension: "plist") else {
+                throw PropertyListConfigurationError.defaultPropertyListNotFound
+            }
```

Comment:
extension on `Bundle`?

Reply:
Since this is associated with the struct itself, it made sense to me to make this an initializer on the configuration object itself, instead of separating it out to a different extension.